### PR TITLE
small updates to auth docs

### PR DESCRIPTION
**Pull request by choldgraf**
Makes GitHub first (since it's the easiest+most free of the auth steps) and adds a bit of extra information

### CODE REVIEW

File: doc/source/authentication.rst [hunk 1]
```diff
@@ -55,6 +49,30 @@ The org_whitelist is optional and will require the use to accept the read:org gi
           org_whitelist:
             - "SomeOrgName"
 
+
+.. note::
+
+   The org_whitelist is optional and will require users to accept the read:org github oath scope when logging in.
```
Comment (manics) [inline doc/source/authentication.rst]: Typo in `oath`

**Comment by choldgraf** [comment]
thanks for noticing @manics !!

File: doc/source/authentication.rst [hunk 2]
```diff
@@ -52,7 +52,8 @@ Below is the structure to use in order to authenticate with GitHub.
 
 .. note::
 
-   The org_whitelist is optional and will require users to accept the read:org github oath scope when logging in.
+   The org_whitelist is optional and will require users to accept the read:org github OAuth scope when
+   logging in.
```
Comment (cam72cam) [inline doc/source/authentication.rst]: Perhaps a note here that it is not required if all the users membership is public

**Comment by choldgraf** [comment]
great point @cam72cam !

**Comment by cam72cam** [comment]
That does not exactly address my concern, perhaps something like:
```
The ``org_whitelist`` section is optional, it lets you only allow users from particular GitHub organizations that you list. If you do not add the auth scope read:org, only users with public membership in the org will be allowed to log in. If it is *not* included, then all GitHub users will be allowed to access your JupyterHub.
```

**Comment by choldgraf** [comment]
@cam72cam hmm - I'm actually a bit unclear as to what `read:org` means, and wasn't able to find options for this when creating a github OAuth app. Could you link to docs on what you're talking about here? Or more generally let me know what this means in more lay-person's terms?

**Comment by manics** [comment]
@choldgraf I've got a related PR open: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/523
Assuming this goes in I could either open a separate docs PR discussing `org_whitelist` and `scope` once this PR is merged in which case just leave your changes as they are, or I could give you a commit to cherry-pick here?

**Comment by choldgraf** [comment]
Let's leave this open and you're welcome to suggest language here that I can add once you merge it in! That work for you?

**Comment by choldgraf** [comment]
(just please remember to ping me when it's merged)

**Comment by cam72cam** [comment]
@choldgraf read:org means that the application can lookup what organizations the user is a member of (both public and private memberships)

**Comment by manics** [comment]
@choldgraf Suggested text in https://github.com/choldgraf/zero-to-jupyterhub-k8s/pull/1

**Pull request merged**